From 0aa566a31a6f376144e36071862e985d1d80a2a0 Mon Sep 17 00:00:00 2001
From: Gustavo Duarte <gduarte@redhat.com>
Date: Fri, 1 Aug 2014 00:50:23 -0400
Subject: [powerpc] perf: Clean up the EBB hash defines a little

Message-id: <1406854226-17977-3-git-send-email-gduarte@redhat.com>
Patchwork-id: 86924
O-Subject: [RHEL7.1 PATCH BZ 1121408 2/5] powerpc/perf: Clean up the EBB hash defines a little
Bugzilla: 1121408
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1121408
Upstream Status: fb568d763f95d356087b447398382b76592ff915

commit fb568d763f95d356087b447398382b76592ff915
Author: Michael Ellerman <mpe@ellerman.id.au>
Date:   Fri Mar 14 16:00:31 2014 +1100

    powerpc/perf: Clean up the EBB hash defines a little

    Rather than using PERF_EVENT_CONFIG_EBB_SHIFT everywhere, add an
    EVENT_EBB_SHIFT like every other event and use that.

    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/perf/power8-pmu.c b/arch/powerpc/perf/power8-pmu.c
index bc8979d..19da3c8 100644
--- a/arch/powerpc/perf/power8-pmu.c
+++ b/arch/powerpc/perf/power8-pmu.c
@@ -88,6 +88,7 @@
  */
 
 #define EVENT_EBB_MASK  1ull
+#define EVENT_EBB_SHIFT  PERF_EVENT_CONFIG_EBB_SHIFT
 #define EVENT_THR_CMP_SHIFT 40 /* Threshold CMP value */
 #define EVENT_THR_CMP_MASK 0x3ff
 #define EVENT_THR_CTL_SHIFT 32 /* Threshold control value (start/stop) */
@@ -120,7 +121,7 @@
   (EVENT_UNIT_MASK      << EVENT_UNIT_SHIFT)  | \
   (EVENT_COMBINE_MASK   << EVENT_COMBINE_SHIFT)  | \
   (EVENT_MARKED_MASK    << EVENT_MARKED_SHIFT)  | \
-  (EVENT_EBB_MASK       << PERF_EVENT_CONFIG_EBB_SHIFT) | \
+  (EVENT_EBB_MASK       << EVENT_EBB_SHIFT)  | \
    EVENT_PSEL_MASK)
 
 /* MMCRA IFM bits - POWER8 */
@@ -236,10 +237,10 @@ static int power8_get_constraint(u64 event, unsigned long *maskp, unsigned long
  pmc   = (event >> EVENT_PMC_SHIFT)        & EVENT_PMC_MASK;
  unit  = (event >> EVENT_UNIT_SHIFT)       & EVENT_UNIT_MASK;
  cache = (event >> EVENT_CACHE_SEL_SHIFT)  & EVENT_CACHE_SEL_MASK;
- ebb   = (event >> PERF_EVENT_CONFIG_EBB_SHIFT) & EVENT_EBB_MASK;
+ ebb   = (event >> EVENT_EBB_SHIFT)        & EVENT_EBB_MASK;
 
  /* Clear the EBB bit in the event, so event checks work below */
- event &= ~(EVENT_EBB_MASK << PERF_EVENT_CONFIG_EBB_SHIFT);
+ event &= ~(EVENT_EBB_MASK << EVENT_EBB_SHIFT);
 
  if (pmc) {
   if (pmc > 6)
-- 
1.7.1